src/Canvas.ts

```diff
@@ -0,0 +1,1011 @@
+   * of the *em* square in the line box, in CSS pixels.
+   */
+  readonly emHeightDescent: number
+
+  /**
+   * The horizontal line indicated by the `text-baseline` property to the hanging baseline of the
+   * line box in CSS pixels.
+   */
+  readonly hangingBaseline: number
+
+  /**
+   * The distance from the horizontal line indicated by the `text-baseline` property to the alphabetic
+   * baseline of the line box in CSS pixels.
+   */
+  readonly alphabeticBaseline: number
+
+  /**
+   * The distance from the horizontal line indicated by the `text-baseline` property to the ideographic
+   * baseline of the line box in CSS pixels.
+   */
+  readonly ideographicBaseline: number
+}
+
+/**
+ * **[UNSAFE]** Gets a canvas element by id.
+ *
+ * @since 1.0.0
+ */
+export const unsafeGetCanvasElementById: (id: string) => HTMLCanvasElement = (id) =>
+  document.getElementById(id) as HTMLCanvasElement
+
+/**
+ * **[UNSAFE]** Gets the 2D graphics context for a canvas element.
+ *
+ * @since 1.0.0
+ */
+export const unsafeGetContext2D: (canvas: HTMLCanvasElement) => CanvasRenderingContext2D = (c) =>
+  c.getContext('2d') as CanvasRenderingContext2D
+
+/**
+ * Gets an canvas element by id, or `None` if the element does not exist or is not an
+ * instance of `HTMLCanvasElement`.
+ *
+ * @since 1.0.0
+ */
+export const getCanvasElementById: (id: string) => IO.IO<O.Option<HTMLCanvasElement>> = (id) => () => {
+  const canvas = unsafeGetCanvasElementById(id)
+  return canvas instanceof HTMLCanvasElement ? O.some(canvas) : O.none
+}
+
+/**
+ * Gets the 2D graphics context for a canvas element.
+ *
+ * @since 1.0.0
+ */
+export const getContext2D: (canvas: HTMLCanvasElement) => IO.IO<CanvasRenderingContext2D> = (c) =>
+  IO.of(unsafeGetContext2D(c))
+
+/**
+ * Gets the canvas width in pixels.
+ *
+ * @since 1.0.0
+ */
+export const getWidth: (canvas: HTMLCanvasElement) => IO.IO<number> = (c) => IO.of(c.width)
+
+/**
+ * Gets the canvas height in pixels.
+ *
+ * since 1.0.0
+ */
+export const getHeight: (canvas: HTMLCanvasElement) => IO.IO<number> = (c) => IO.of(c.height)
+
+/**
+ * Sets the width of the canvas in pixels.
+ *
+ * @since 1.0.0
+ */
+export const setWidth: (width: number) => (canvas: HTMLCanvasElement) => IO.IO<void> = (w) => (c) => () => {
+  c.width = w
+}
+
+/**
+ * Sets the height of the canvas in pixels.
+ *
+ * @since 1.0.0
+ */
+export const setHeight: (height: number) => (canvas: HTMLCanvasElement) => IO.IO<void> = (h) => (c) => () => {
+  c.height = h
+}
+
+/**
+ * Gets the dimensions of the canvas in pixels.
+ *
+ * @since 1.0.0
+ */
+export const getDimensions: (canvas: HTMLCanvasElement) => IO.IO<CanvasDimensions> = (c) =>
+  sequenceS(IO.io)({ height: getHeight(c), width: getWidth(c) })
+
+/**
+ * Sets the dimensions of the canvas in pixels.
+ *
+ * @since 1.0.0
+ */
+export const setDimensions: (dimensions: CanvasDimensions) => (canvas: HTMLCanvasElement) => IO.IO<void> = (d) => (c) =>
```

Comment:
idem, should return `IO.IO<HTMLCanvasElement>`

Reply:
https://github.com/gcanti/graphics-ts/pull/7/commits/7d16456374e7b1649daca1088a9035c88a3605dd